### PR TITLE
CIR-1351-Implement Navigation changes for setion 5 and validation for…

### PR DESCRIPTION
**Pull request by AdamCoops**
… greater thans

### CODE REVIEW

File: app/navigation/UkCompaniesNavigator.scala [hunk 1]
```diff
@@ -42,17 +42,21 @@ class UkCompaniesNavigator @Inject()() extends Navigator {
...
       case None         => routes.AddRestrictionController.onPageLoad(idx, NormalMode)
     }),
-    CompanyAccountingPeriodSameAsGroupPage -> ((_, _) => controllers.routes.UnderConstructionController.onPageLoad()), //TODO: Update as part of routing subtask
-    RestrictionAmountSameAPPage -> ((_, _) => controllers.routes.UnderConstructionController.onPageLoad()), //TODO: Update as part of routing subtask
+    CompanyAccountingPeriodSameAsGroupPage -> ((idx, userAnswers) => userAnswers.get(UkCompaniesPage, Some(idx)).flatMap(_.accountPeriodSameAsGroup) match {
+      case Some(true) => controllers.ukCompanies.routes.RestrictionAmountSameAPController.onPageLoad(idx, NormalMode)
+      case Some(false) => controllers.routes.UnderConstructionController.onPageLoad()
+      case _ => routes.CompanyAccountingPeriodSameAsGroupController.onPageLoad(idx, NormalMode)
+    }),
+    RestrictionAmountSameAPPage -> ((idx, _) => controllers.ukCompanies.routes.CompanyEstimatedFiguresController.onPageLoad(idx, NormalMode)),
```
Comment (burns238) [inline app/navigation/UkCompaniesNavigator.scala:59]: This actually needs to be CompanyContainsEstimatesController

File: conf/messages.en [hunk 2]
```diff
@@ -892,6 +892,7 @@ restrictionAmountSameAP.error.nonNumeric = Enter the restriction amount using nu
 restrictionAmountSameAP.error.required = Enter the restriction amount for this company
 restrictionAmountSameAP.error.invalidNumeric = Restriction amount must be to 2 decimal places or a whole number
 restrictionAmountSameAP.error.outOfRange = Restriction amount must be between {0} and {1}
+restrictionAmountSameAP.error.expenseAmount = A company cannot have a restriction greater than it's next tax expense
```
Comment (burns238) [inline conf/messages.en:895]: Should this be "net tax expense"?
And shout "it's" be "its"

**Review by burns238 — CHANGES_REQUESTED** [review]

File: app/navigation/UkCompaniesNavigator.scala (hunk 1 above)
Comment (AdamCoops) [reply app/navigation/UkCompaniesNavigator.scala:59]: What leads to CompanyContainsEstimatesPage then?

File: conf/messages.en (hunk 2 above)
Comment (AdamCoops) [reply conf/messages.en:898]: "A company cannot have a restriction greater than it's next tax expense"
Is the exact wording in the user story.

**Review by burns238 — APPROVED** [review]

**Pull request merged**